Python/Product/VSInterpreters/Interpreter/CondaEnvironmentFactoryProvider.cs

```diff
@@ -96,28 +101,22 @@ class CondaEnvironmentFactoryProvider : IPythonInterpreterFactoryProvider, IDisp
                     _initialized = true;
                     doDiscover = true;
                     try {
-                        _environmentsTxtPath = Path.Combine(
+                        _environmentsTxtFolder = Path.Combine(
                             Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
```

Comment:
Should we make this a parameter and add a test for this scenario? (So we don't have to mess with `UserProfile` on the test machines)

Reply:
I don't want it to mess with the folder on my machine either.

I'd be surprised if we can tell conda to use a different path though, I'll have to check.

I could easily enough make the factory provider use an alternate path, have the test create the environments.txt file in that path, and then check that this triggered a rediscovery. There won't be a InterpretersChanged event triggered because `conda list` will return the same results as before. But I could add an internal DiscoveryStarted event and have the test expect that instead.